workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs

```diff
@@ -160,5 +164,26 @@ private void DrawCodeGenerationOptions()
 
             EditorGUIUtility.labelWidth = previousWidth;
         }
+
+        private void DrawCustomSnapshotDir()
+        {
+            GUILayout.Label(CustomSnapshotPathLabel, EditorStyles.boldLabel);
+            GUILayout.Space(EditorGUIUtility.standardVerticalSpacing);
+
+            using (new EditorGUILayout.HorizontalScope())
+            {
+                EditorGUI.BeginDisabledGroup(true);
+                EditorGUILayout.TextField(CustomSnapshotPathLabel, toolsConfig.CustomSnapshotPath);
+                EditorGUI.EndDisabledGroup();
+                if (GUILayout.Button("Open", EditorStyles.miniButtonRight, GUILayout.ExpandWidth(false)))
+                {
+                    var path = EditorUtility.OpenFilePanel("Select snapshot", Path.GetFullPath(toolsConfig.CustomSnapshotPath), "snapshot");
+                    if (path.Length != 0)
```

Comment:
Can do `!string.IsNullOrEmpty(path)`. This is typically our style and guards against `null`s as well 😄